McEngine/src/GUI/CBaseUICanvas.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * CBaseUICanvas.h
+ *
+ *  Created on: May 28, 2017
+ *      Author: Psy
+ */
+
+#ifndef GUI_WINDOWS_CBASEUICANVAS_H_
+#define GUI_WINDOWS_CBASEUICANVAS_H_
+
+#include "cbase.h"
+#include "Engine.h"
+#include "CBaseUIElement.h"
+
+/*
+ * UI Canvas Container
+ * Scales any slotted containers or elements by the size of the canvas, useful for resolution scaling
+ * The size/position of UI elements slotted should 0.0 to 1.0 as a percentage of the total screen area
+ * Set scaleByHeightOnly per element to avoid stretching/squashing on aspect ratio changes. Uses a 16:9 (Widescreen) aspect ratio for assumed desired width
+ */
+struct Slot{

```

Comment:
Be careful with declaring generic names in the global namespace. Make the Slot struct a public member of CBaseUICanvas, and access it via CBaseUICanvas::Slot